pkg/manager-nvme/nvme_cli.go

```diff
@@ -278,6 +278,51 @@ func (d *cliDevice) DeleteNamespace(namespaceId nvme.NamespaceIdentifier) error
 	return nil
 }
 
+func (d *cliDevice) FormatNamespace(namespaceID nvme.NamespaceIdentifier) error {
+	// Example Command
+	//    # nvme format --force --namespace-id=1 /dev/nvme2
+	//    format: Success, formatted nsid:1
+
+	rsp, err := d.run(fmt.Sprintf("format %s --force --namespace-id=%d", d.dev(), namespaceID))
+	if err != nil {
+		return err
+	}
+
+	regex := regexp.MustCompile(`^format: Success, formatting nsid:(?P<NSID>\d+)\n`)
+	substrings := regex.FindStringSubmatch(rsp)
```

Comment:
What is `format` outputting that we're using to search for a substring here? And why should it return 2 of them?

Reply:
We are trying to catch the substrings: 'Success formatting namespace:' and 'namespace-number' as substrings.
The 2nd should match the namespace-id that we input.

Reply:
regex always returns the full match (if present) first, then the submatch[1] is the NSID